krun/config.py

```diff
@@ -41,16 +41,13 @@ def _fatal_exception_execing_config(self, exc_info):
             lines.append(frame)
         fatal("".join(lines))
 
-    def read_from_string(self, config_str):
-        config_dict = {}
-        try:
-            exec(config_str, config_dict)
-        except Exception as e:
-            self._fatal_exception_execing_config(sys.exc_info())
-
-        self.__dict__.update(config_dict)
-        self.filename = ""
-        self.text = config_str
+    def check_config_consistency(self, config_str):
+        import difflib
+        if self.text != config_str:
+            diff = "\n".join(difflib.unified_diff(self.text.splitlines(),
+                                                  config_str.splitlines()))
+            fatal("the config file %s on disk was different than "
+                  "the resumed config file:\n%s" % (self.filename, diff))
```

Comment:
You are right, we could be more precise. I'm trying to give an error message that will also make sense to a user who doesn't know that we cache the config file (I had forgotten, for example).

How about:

```
The experiment is in an inconsistent state as config file x.krun has changed since it was initially cached in x_results.json.bz2
```

(Maybe I am over-thinking)